data/db-router.js

```diff
@@ -0,0 +1,98 @@
+const router = require('express').Router();
+
+const db = require('./db.js');
+
+router.post('/', (req, res) => {
+  const newPost = req.body;
+  if (newPost.title && newPost.contents) {
+    db.insert(newPost)
+      .then(post => {
+        res.status(201).json(newPost);
+      })
+      .catch(err => {
+        res.status(500).json({
+          error: 'There was an error while saving the post to the database'
+        });
+      });
+  } else {
+    res.status(400).json({
+      errorMessage: 'Please provide title and contents for the post.'
+    });
+  }
+});
+
+router.get('/', (req, res) => {
+  db.find()
+    .then(posts => {
+      res.status(200).json(posts);
+    })
+    .catch(err => {
+      res
+        .status(500)
+        .json({ error: 'The posts information could not be retrieved.' });
+    });
+});
+
+router.get('/:id', (req, res) => {
+  db.findById(req.params.id)
+    .then(post => {
+      if (post[0]) {
```

Comment:
there's a couple other ways you could accomplish this with destructuring or just checking the `post.length`, but this will also work. this could also be handled from the helper functions (when we get to the point when you're writing all the SQL side of things) there's actually a method called `.first()` which automatically returns the first index item in a returned array even if it's only one item